### PR TITLE
simplify GitHub Actions

### PR DESCRIPTION
**Pull request by simonbyrne**
Will need to update branch protections to merge this

- [ ] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: .github/workflows/UnitTests.yml [hunk 1]
```diff
@@ -36,5 +43,6 @@ jobs:
       uses: julia-actions/julia-buildpkg@v1
 
     - name: Run Unit Tests
-      run: julia --project -e 'using Pkg; Pkg.test()'
-      continue-on-error: ${{ matrix.version == 'nightly' }}
+      uses: julia-actions/julia-runtest@v1
+      with:
+        annotate: true
```
Comment (Sbozzolo) [inline .github/workflows/UnitTests.yml:48]: Can you please add a carriage return here at the end of this line?

**Comment by Sbozzolo** [comment]
Is the test on Julia 1.10 required to pass?

**Comment by simonbyrne** [comment]
Currently this won't pass: for this to merge we will have to change the required checks.

File: .github/workflows/UnitTests.yml (hunk 1 above)
Comment (simonbyrne) [reply .github/workflows/UnitTests.yml:48]:
```suggestion
        annotate: true

```

**Comment by simonbyrne** [comment]
But to actually answer your question: we can just specify which ones we want to pass (so I can make it just the 1.9 ones for now)

**Review by Sbozzolo — APPROVED** [review]

**Comment by simonbyrne** [comment]
I've changed the required checks, so this one needs to get merged before any others

**Pull request merged**